File: CBLClient/Apps/JavaRequestHandlers/src/main/java/com/couchbase/mobiletestkit/javacommon/RequestHandler/DataTypesInitiatorHandler.java
package com.couchbase.mobiletestkit.javacommon.RequestHandler;

import java.util.Date;

import com.couchbase.mobiletestkit.javacommon.Args;


public class DataTypesInitiatorHandler {
    /* ---------------------------------- */
    /* - Initiates Complex Java Objects - */
    /* ---------------------------------- */


    public Date setDate(Args args) {
        return new Date();
    }

    public Double setDouble(Args args) {
        return Double.parseDouble(args.getString("value"));
    }

    public Float setFloat(Args args) {
        return Float.parseFloat(args.getString("value"));
    }

    public Long setLong(Args args) {
        return Long.parseLong(args.getString("value"));
    }

    public Boolean compare(Args args) {
        String first = args.getString("first");
        String second = args.getString("second");
        return first.equals(second);
    }

    public Boolean compareDate(Args args) {
        Date first = args.get("date1", Date.class);
        Date second = args.get("date2", Date.class);
        return first.equals(second);
    }

    public Boolean compareDouble(Args args) {
        Double first = Double.valueOf(args.getString("double1"));
        Double second = Double.valueOf(args.getString("double2"));
        return first.equals(second);
    }

    public Boolean compareLong(Args args) {
        Long first = args.getLong("long1");
        Long second = args.getLong("long2");
        return first.equals(second);
    }

}

File: CBLClient/Apps/JavaRequestHandlersKotlin/src/main/java/com/couchbase/mobiletestkit/javacommon/RequestHandler/DataTypesInitiatorHandler.kt
package com.couchbase.mobiletestkit.javacommon.RequestHandler

import com.couchbase.mobiletestkit.javacommon.Args
import java.util.*

class DataTypesInitiatorHandler {
    /* ---------------------------------- */ /* - Initiates Complex Java Objects - */ /* ---------------------------------- */
    fun setDate(args: Args?): Date {
        return Date()
    }

    fun setDouble(args: Args): Double {
        return args.get<Any>("value").toString().toDouble()
    }

    fun setFloat(args: Args): Float {
        return args.get<Any>("value").toString().toFloat()
    }

    fun setLong(args: Args): Long {
        return args.get<Any>("value").toString().toLong()
    }

    fun compare(args: Args): Boolean {
        val first = args.get<Any>("first").toString()
        val second = args.get<Any>("second").toString()
        return first == second
    }

    fun compareDate(args: Args): Boolean {
        val first = args.get<Date>("date1")
        val second = args.get<Date>("date2")
        return first == second
    }

    fun compareDouble(args: Args): Boolean {
        val first = java.lang.Double.valueOf(args.get<Any>("double1").toString())
        val second = java.lang.Double.valueOf(args.get<Any>("double2").toString())
        return first == second
    }

    fun compareLong(args: Args): Boolean {
        val first = args.get<Long>("long1")
        val second = args.get<Long>("long2")
        return first == second
    }
}